### PR TITLE
Feature/fix alloc di alloc consistency

### PR DESCRIPTION
**Pull request by Khalifa1997**

### CODE REVIEW

File: src/shogun/features/SNPFeatures.cpp [hunk 1]
```diff
@@ -351,7 +351,7 @@ std::shared_ptr<Features> SNPFeatures::duplicate() const
 SGMatrix<float64_t> SNPFeatures::get_histogram(bool normalize)
 {
 	int32_t nsym=3;
-	float64_t* h= SG_CALLOC(float64_t, size_t(nsym)*string_length/2);
+	SGMatrix<float64_t> h(nsym, string_length/2,false);
```
Comment (vigsterkr) [inline src/shogun/features/SNPFeatures.cpp]: why false?

Comment (vigsterkr) [inline src/shogun/features/SNPFeatures.cpp]:
```suggestion
	SGMatrix<float64_t> h(nsym, string_length/2);
```

File: src/shogun/features/StringFeatures.cpp [hunk 2]
```diff
@@ -160,10 +160,10 @@ template<class ST> SGVector<ST> StringFeatures<ST>::get_feature_vector(int32_t n
 	int32_t l;
 	bool free_vec;
 	ST* vec=get_feature_vector(num, l, free_vec);
-	ST* dst=SG_MALLOC(ST, l);
-	sg_memcpy(dst, vec, l*sizeof(ST));
+	SGVector<ST>dst(l);
```
Comment (vigsterkr) [inline src/shogun/features/StringFeatures.cpp]: plz use spaces (still) :)

**Review by vigsterkr — CHANGES_REQUESTED** [review]

File: src/shogun/features/StringFeatures.cpp (hunk 2 above)
Comment (Khalifa1997) [reply src/shogun/features/StringFeatures.cpp]: haha okay

File: src/shogun/features/SNPFeatures.cpp (hunk 1 above)
Comment (vigsterkr) [reply src/shogun/features/SNPFeatures.cpp]:
```suggestion
	SGMatrix<float64_t> h(nsym, string_length/2);
```

Comment (Khalifa1997) [reply src/shogun/features/SNPFeatures.cpp]: nvm must of copied that one by mistake sorry :D

**Pull request merged**